app/dao/invited_user_dao.py

```diff
@@ -13,32 +13,28 @@ def get_invited_user_by_service_and_id(service_id, invited_user_id):
     return InvitedUser.query.filter(
         InvitedUser.service_id == service_id,
         InvitedUser.id == invited_user_id,
-        InvitedUser.status != INVITE_EXPIRED,
     ).one()
 
 
 def get_invited_user_by_id(invited_user_id):
-    return InvitedUser.query.filter(
-        InvitedUser.id == invited_user_id, InvitedUser.status != INVITE_EXPIRED
-    ).one()
+    return InvitedUser.query.filter(InvitedUser.id == invited_user_id).one()
 
 
 def get_expired_invited_users_for_service(service_id):
-    return InvitedUser.query.filter(
-        InvitedUser.service_id == service_id, InvitedUser.status == INVITE_EXPIRED
-    ).all()
+    return InvitedUser.query.filter(InvitedUser.service_id == service_id).all()
 
 
 def get_invited_users_for_service(service_id):
-    return InvitedUser.query.filter(
-        InvitedUser.service_id == service_id, InvitedUser.status != INVITE_EXPIRED
-    ).all()
+    return InvitedUser.query.filter(InvitedUser.service_id == service_id).all()
```

Comment:
This method and the one above it now do the same thing as they are not filtering on a status; am I correct in that this is what will ensure that we see all invites in the admin interface now instead of just the non-expired invites?

If this is the case, a couple of things cross my mind:

- Did these need to change at all, or did the admin interface just need to query explicitly to show expired invites (which may have required some adjustments or additions to `app/notify_client/invite_api_client.py` in the admin site)?
- If this is an appropriate change to make, would it make sense to consolidate these into a single method and use that going forward, e.g., just sticking with this one since it's more accurately named for what's happening now?

This also leads me to another question after looking things over:  what kind of impact would this have on some of the other existing routes that are set up for looking at invites with a specific status in `app/service_invite/rest.py`, e.g., `/service/<service_id>/invite/expired`?  They wouldn't return the results that are currently expected, would they?

I did a quick look through the admin codebase (`app/notify_client/invite_api_client.py` in particular) and do not see these endpoints used, so it's likely a non-issue at the moment, but when we start to expose the API itself in the future we'd likely have to account for this.  Hence my reason for asking! :-)

Reply:
The expired endpoint was one I made in 96, and I found it was unnecessary honestly. It could be safely removed.

When digging through the code and seeing how the statuses were being used, I felt confident that the change I did in 96 to block "expired" from being shown, could be reversed - because when that list of invites was used, the status was already being checked to see if it was a specific status (like "pending"), to filter out ones like "cancelled" or "accepted". So what this will do is list the expired invites in the list of invites, so then that expired invite can be selected and then re-sent.

Further, when running through the tests, all of them worked even after I did this change back - so to me, it seemed like a bit of over-engineering to explicitly filter out the expired invites from the list of invites.

I wanted the expired invites to load into the main list of invites so that the changes to the UI would be minimal (no need to make an additional query to explicitly get those to add them to the list of invites on the UI).

I agree that there will need to be full testing on the frontend to ensure that it works as expected afterward.

Reply:
Ah okay!  I forgot about the addition in 96, that's what confused me here.  Thanks for explaining!

This makes sense then, and from the UI perspective too, so if anything we could just remove the other method(s) for the expired endpoint?